cloud_Azure/terraform/module/examples/multiple_accounts_multiple_resource_group/profiles_populator.py

```diff
@@ -0,0 +1,479 @@
+            "Failed to consent permissions for Service Principal '%s' in Subscription '%s'", name, subscription_id
+        )
+        return None
+
+    log.info("Configured Service Principal '%s' with ID '%s' in Subscription '%s'", name, principal.id, subscription_id)
+    return principal
+
+
+def list_locations() -> List[str]:
+    """
+    Return currently available Azure locations on success
+    Return empty list on error
+    """
+
+    COMMAND = "account list-locations --query '[].name'"  # returns a list
+    output_list = az_cli(COMMAND)
+    if not isinstance(output_list, list):
+        log.error("Failed to list Azure Locations")
+        return []
+
+    log.info("Retrieved %d Azure Locations", len(output_list))
+    return sorted(output_list)
+
+
+def list_resource_groups(location: str) -> List[str]:
+    """
+    Return Resource Groups available in "location" on success
+    Return empty list on error
+    """
+
+    if not location:
+        log.warning("Location not specified, returning empty Resource Group list")
+        return []
+
+    COMMAND = f'''group list --query "[?location=='{location}'].name"'''  # returns a list
+    output_list = az_cli(COMMAND)
+    if not isinstance(output_list, list):
+        log.error("Failed to list Resource Groups in location '%s'", location)
+        return []
+
+    log.info("Found %d Resource Group(s) in location '%s'", len(output_list), location)
+    return output_list
+
+
+def get_service_principal(name: str, profiles: List[AzureProfile]) -> Optional[ServicePrincipal]:
+    principal_id = find_service_principal(name)
+    if not principal_id:
+        return None
+    principal_secret = find_secret(principal_id, profiles)
+    return ServicePrincipal(id=principal_id, secret=principal_secret or "")
+
+
+def find_service_principal(name: str) -> Optional[str]:
+    """
+    Return principal_id on success
+    Return empty string if principal not found
+    Return None on lookup error
+    """
+
+    COMMAND = f"""ad app list --query "[?displayName == '{name}'].appId" """  # returns a list
+    output_list = az_cli(COMMAND)
+    if not isinstance(output_list, list):
+        log.error("Failed to lookup Service Principal '%s'", name)
+        return None
+    if output_list == []:
+        log.info("Service Principal '%s' not found", name)
+        return ""
+
+    principal_id = output_list[0]
+    log.info("Service Principal '%s' found with ID '%s'", name, principal_id)
+    return principal_id
+
+
+def new_service_principal(subscription_id: str, name: str) -> Optional[ServicePrincipal]:
+    """
+    Return ServicePrincipal on success
+    Return None on error
+    """
+
+    COMMAND = f"""ad sp create-for-rbac --role="Owner" --scopes="/subscriptions/{subscription_id}" --name {name}"""  # returns a dict
+    output_dict = az_cli(COMMAND)
+    if not isinstance(output_dict, dict):
+        log.error("Failed to create Service Principal '%s' in Subscription '%s'", name, subscription_id)
```

Comment:
This log duplicates the one from parent function (L207)

Reply:
Removed duplication